### ISSUE TITLE
Add streaming JSON output support

### ISSUE THREAD

**Issue by singpolyma** [kellyjonbrazil/jc#157]
So for example `jc -s ls` could emit:

```
{"filename":"file1"}{"filename":"file2"}
```

Instead of making one big array.  This allows tools like jq to process incrementally for large lists.

**Comment by kellyjonbrazil** [comment]
Thank you for the suggestion!

I have been toying with the idea of supporting something like this - probably JSON Lines. I'm thinking I want to not only output a stream but also input and process as a stream. This will be a bit of an undertaking so I'll need to put some thought into the implementation. I think it could be a nice feature, though! I might target this for v1.17.0 if I can find a good way to implement this.

**Comment by singpolyma** [comment]
Emitting JSON lines would be fine, though a bit verbose.  For parsing, not sure what JSON parsers are available to you but would be ideal if it could handle arbitrarily streamed JSON like jq can.

**Comment by kellyjonbrazil** [comment]
I have a working prototype of a streaming `ls` command parser in the [`streaming`](https://github.com/kellyjonbrazil/jc/tree/streaming) branch:
https://github.com/kellyjonbrazil/jc/blob/streaming/jc/parsers/ls_s.py

**Comment by kellyjonbrazil** [comment]
I made skipping exceptions optional by using the `-q` or `quiet` option. Not sure if this is the best way forward, but seems sorta logical.
```
$ ping 1.1.1.1 | jc --ls-s
jc:  Error - Parser issue with ls-s:
             Unparsable line: PING 1.1.1.1 (1.1.1.1): 56 data bytes

             For details use the -d or -dd option. Use "jc -h" for help.
```
Using `-q`:
```
$ ping 1.1.1.1 | jc --ls-s -q
{"_meta":{"success":false,"error":"error parsing line","line":"PING 1.1.1.1 (1.1.1.1): 56 data bytes"}}
{"_meta":{"success":false,"error":"error parsing line","line":"64 bytes from 1.1.1.1: icmp_seq=0 ttl=56 time=26.849 ms"}}
...
```

**Comment by kellyjonbrazil** [comment]
I've completed the streaming parser functionality in the [dev](https://github.com/kellyjonbrazil/jc/tree/dev) branch. Two streaming parsers are included: `ls-s` and `ping-s`.

Also, added support for unbuffering output using the `-u` option. This is helpful for streaming output from `ping`, which doesn't output enough data fast enough to fill up the pipe buffers if you are piping from `jc` to something else, like `jq`, and printing to the terminal.

I'm still working on a couple more things, but I think v1.17.0 should be released within the next couple weeks with this functionality and a couple other things.

**Comment by kellyjonbrazil** [comment]
Streaming parser support is now available in v1.17.0.
